### PR TITLE
Data PreProcessing into SegNet input format

### PR DESCRIPTION
**Pull request by hackgoofer**
To run:
`python3 image_squarizer_sanitizer.py -i ../Pictures/cat+face -o ../Pictures/output -s 256`

Before this, I used image_size_inspector to sanity check that all images are within the 256 height and weight, and removed the ones that don't fit the size requirements

And I ran image_squarizer_sanitizer to get the images and their masks to the expected input format that SetNet needs.

### CODE REVIEW

File: bin/image_size_inspector.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+from PIL import Image
+import os
+
+def get_num_pixels(file_path):
+  width, height = Image.open(file_path).size
+  return (width, height)
+
+def analyze_directory(directory_path):
+  directory = os.fsencode(directory_path)
```
Comment (uncommoncode) [inline bin/image_size_inspector.py]: Is the `os.fsencode` necessary with `os.listdir`? I think it is only needed for command line arguments

File: bin/image_squarizer_sanitizer.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  newfilepath = os.path.join(directory_path, newfilename)
+  print("Saving: " + newfilename)
+  try:
+    new_im.save(newfilepath)
+  except IOError:
+    print("Error: cannot save for" + newfilepath)
+
+def binarizeImage(img, size, img_path):
+  pix = img.load()
+  for x in range(size[0]):
```
Comment (uncommoncode) [inline bin/image_squarizer_sanitizer.py]: Numpy provides a nice way to do this kind of stuff. You can write this sort of clamping code like: `pix[pix > 0] = 1`

If `pix` is not a numpy matrix, then you can convert with `pix = np.array(pix)`

File: bin/image_squarizer_sanitizer.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  new_im = new_im.resize((new_size, new_size), Image.NEAREST)
+  newfilename = os.path.splitext(filename)[0] + "_new.jpg"
+  newfilepath = os.path.join(directory_path, newfilename)
+  print("Saving: " + newfilename)
+  try:
+    new_im.save(newfilepath)
+  except IOError:
+    print("Error: cannot save for" + newfilepath)
+
+def binarizeImage(img, size, img_path):
```
Comment (uncommoncode) [inline bin/image_squarizer_sanitizer.py]: should probably be PEP-8 style binarize_image

File: bin/image_squarizer_sanitizer.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+
+from PIL import Image
+import os
+import argparse
+
+OUTPUT = os.path.join(os.path.dirname(os.getcwd()), "Pictures", "output")
+INPUT = os.path.join(os.path.dirname(os.getcwd()), "Pictures", "cat+face")
+
+class readable_dir(argparse.Action):
```
Comment (uncommoncode) [inline bin/image_squarizer_sanitizer.py:16]: This is cool!!

File: bin/image_squarizer_sanitizer.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        print("Padding image and map: " + file_name_wo_extension)
+        img = Image.open(file_path)
+        msk = Image.open(map_file_path)
+        size = (max(img.size), max(img.size))
+        pad_image(size, new_size, img, file_name, output_path)
+        binarizeImage(msk, msk.size, map_file_path)
+        pad_image(size, new_size, msk, map_name, output_path)
+    
+def pad_image(size, new_size, img, filename, directory_path):
+  new_im = Image.new("RGB", size)
```
Comment (uncommoncode) [inline bin/image_squarizer_sanitizer.py]: We probably want to have the image background be white noise. This looks like a reasonable way to do it with PIL: https://gist.github.com/glenrobertson/2288152#file-white_noise_image-py

File: bin/image_squarizer_sanitizer.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  new_im.paste(img, (0, 0))
+  # Question: not sure if nearest is the best option here
```
Comment (uncommoncode) [inline bin/image_squarizer_sanitizer.py]: I don't think it should matter too much but I would use the default `Image.BICUBIC`.

**Review by uncommoncode — APPROVED** [review]
Looking good!!

File: bin/image_size_inspector.py (hunk 1 above)
Comment (hackgoofer) [reply bin/image_size_inspector.py]: you are correct.

**Comment by hackgoofer** [comment]
Updated with the comments.

**Pull request merged**